scalameta/contrib/src/main/scala/scala/meta/contrib/ScaladocParser.scala

```diff
@@ -0,0 +1,120 @@
+package scala.meta.contrib
+
+import DocToken._
+import scala.meta.Syntax
+import scala.meta.tokens.Token.Comment
+
+object ScaladocParser {
+
+  /**
+    * Parses a scaladoc [[Comment]].
+    */
+  def parseScaladoc(comment: Comment): Seq[DocToken] = {
+
+    val trimmedComment: String = comment.show[Syntax].trim
+
+    require(
+      requirement = trimmedComment.startsWith("/*") && trimmedComment.endsWith("*/"),
+      message = "Input comment is not a Scaladoc"
+    )
+
+    val scaladocSymbols = Seq('/', '*', ' ')
+
+    val lineSeparedDocString: Seq[String] =
+      trimmedComment
+        .split(Array('\n', '\r')) // Splits the Scaladoc in lines
```

Comment:
`.lines` does the same to get an `Iterator[String]`.

I think we should try to avoid parsing line-by-line however, since bodies and descriptions can be multiline. Fastparse is designed for a use-case like this.